src/main/scala/tech/cryptonomic/conseil/config/Security.scala

```diff
@@ -1,8 +1,46 @@
 package tech.cryptonomic.conseil.config
 
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import cats.effect.IO
+import cats.effect.concurrent.Ref
+import com.typesafe.scalalogging.LazyLogging
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport
 import pureconfig.generic.auto._
 
-object Security {
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object Security extends ErrorAccumulatingCirceSupport with LazyLogging {
+
+  /** Keys cache */
+  private val nautilusCloudKeys: Ref[IO, Set[String]] = Ref.of[IO, Set[String]](Set.empty).unsafeRunSync()
+
+  /** Updates API keys from Nautilus-Cloud endpoint */
+  def updateKeys(
+      ncc: NautilusCloudConfiguration
+  )(implicit executionContext: ExecutionContext, system: ActorSystem, mat: Materializer): Unit = {
+    val update = for {
+      apiKeys <- Http()
+        .singleRequest(
+          HttpRequest(uri = makeUri(ncc))
+            .withHeaders(RawHeader("X-Api-Key", ncc.key))
+        )
+        .flatMap(Unmarshal(_).to[Set[String]])
+      _ <- nautilusCloudKeys.update(_ => apiKeys).unsafeToFuture()
+    } yield logger.info(s"Managed to update api keys with $apiKeys")
```

Comment:
I may be wrong, but this line seems to output all the fetched keys into the log. That may not be ideal, especially when we may have many many keys.

Reply:
ok, I found it useful to check if the keys were loaded correctly. What do you think about changing it to `debug` level? Or do you prefer just removing them from the log?

Reply:
If would prefer keys to not be in the log under normal operations, just a one liner saying that the fetch was successful.

If you feel it useful, you can add a secondary debug statement for printing the keys out.

Reply:
ok, it was useful during development, now I removed it